### ISSUE TITLE
[libmysql] Build error

### ISSUE THREAD

**Issue by qlaiaqu** [microsoft/vcpkg#32878]
**Operating system**
Ubuntu 22.04.3 X86_64

**Cross Compiler**
http://releases.linaro.org/components/toolchain/binaries/7.3-2018.05/aarch64-linux-gnu/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu.tar.xz

**Steps to reproduce the behavior**

1. tar -xJvf gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu.tar.xz -C /opt/
2. export PATH=${PATH}:/opt/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu/bin
3. ./vcpkg install libmysql:arm64-linux

**Failure logs**
Computing installation plan...
The following packages will be built and installed:
    libmysql:arm64-linux -> 8.0.32#6
Detecting compiler hash for triplet arm64-linux...
Restored 0 package(s) from /home/qlaiaqu/.cache/vcpkg/archives in 468 us. Use --debug to see more details.
Installing 1/1 libmysql:arm64-linux...
Building libmysql:arm64-linux...
warning: -- Using community triplet arm64-linux. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /home/qlaiaqu/vcpkg-linux/triplets/community/arm64-linux.cmake
CMake Warning at ports/libmysql/portfile.cmake:6 (message):
  'autoconf-archive' must be installed via your system package manager (brew,
  apt, etc.).
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)

-- Using cached mysql-mysql-server-mysql-8.0.32.tar.gz.
./vcpkg install libmysql:arm64-linux-- Cleaning sources at /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/src/sql-8.0.32-820d934206.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/qlaiaqu/vcpkg-linux/downloads/mysql-mysql-server-mysql-8.0.32.tar.gz
-- Applying patch ignore-boost-version.patch
-- Applying patch system-libs.patch
-- Applying patch export-cmake-targets.patch
-- Applying patch Add-target-include-directories.patch
-- Applying patch homebrew.patch
-- Applying patch fix_dup_symbols.patch
-- Using source at /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/src/sql-8.0.32-820d934206.clean
-- Configuring arm64-linux
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /home/qlaiaqu/vcpkg-linux/downloads/tools/ninja/1.10.2-linux/ninja -v
    Working Directory: /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/arm64-linux-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-dbg-CMakeCache.txt.log
      /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-rel-CMakeCache.txt.log
      /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/libmysql/portfile.cmake:40 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)

**Comment by FrankXie05** [comment]
Please provide the error log: 
```
/home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-dbg-CMakeCache.txt.log
/home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-rel-CMakeCache.txt.log
/home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-out.log
```

**Comment by mz02005** [comment]
> > Please provide the error log:
> > ```
> > /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-dbg-CMakeCache.txt.log
> > /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-rel-CMakeCache.txt.log
> > /home/qlaiaqu/vcpkg-linux/buildtrees/libmysql/config-arm64-linux-out.log
> > ```
> 
> Thank you for your answer. I think I found the reason. The compilation process of libmysql generates some intermediate programs. These programs cannot run on the X64 machine because they are based on the ARM architecture family. Because the version of libevent cannot be detected, the compilation process is terminated. I tried to disable the version detection of libevent, but there were still some intermediate programs that could not run, resulting in the compilation process being terminated.
> 
> ` /home/qlaiaqu/vcpkg-linux/installed/arm64-linux/lib/libevent_core.a;/home/qlaiaqu/vcpkg-linux/installed/arm64-linux/lib/libevent_extra.a;LIBEVENT_OPENSSL-NOTFOUND;/home/qlaiaqu/vcpkg-linux/installed/arm64-linux/lib/libevent_pthreads.a CMake Error at cmake/libevent.cmake:164 (MESSAGE): LIBEVENT version must be at least 2.1, found /.
> 
> Please use -DWITH_LIBEVENT=bundled Call Stack (most recent call first): CMakeLists.txt:1881 (MYSQL_CHECK_LIBEVENT) `

so, how did you resolve it? Modified the libmysql port file, add -DWITH_LIBEVENT=bundled to the options?

**Comment by mz02005** [comment]
**I have the same issue. 
I try to modify the libmysql port file, replace '-DWITH_LIBEVENT=system' with '-DWITH_LIBEVENT=bundled'. Then, I got error logs as below:**

...
-- LIBEVENT_VERSION (bundled) 0
-- LIBEVENT_INCLUDE_DIRS /home/mz/works/vcpkg/buildtrees/libmysql/src/sql-8.0.32-820d934206.clean/extra/libevent/libevent-2.1.11-stable/include;/home/mz/works/vcpkg/buildtrees/libmysql/arm64-linux-release-dbg/extra/libevent/libevent-2.1.11-stable/include
-- LIBEVENT_LIBRARIES event_core;event_extra;event_openssl;event_pthreads
CMake Error at cmake/libevent.cmake:164 (MESSAGE):
  LIBEVENT version must be at least 2.1, found 0.

  Please use -DWITH_LIBEVENT=bundled
Call Stack (most recent call first):
  CMakeLists.txt:1881 (MYSQL_CHECK_LIBEVENT)
...

**Comment by Krzmbrzl** [comment]
I also encountered this error on a M1 mac, where I tried to cross-compile for x64-osx.

**Comment by mcginnie** [comment]
I have similar/same problem

===========================================================================
Installing 1/7 libmysql:x64-windows...
Building libmysql:x64-windows...
-- Using cached mysql-mysql-server-mysql-8.0.34.tar.gz.
-- Cleaning sources at D:/Dev/C++/vcpkg/buildtrees/libmysql/src/sql-8.0.34-ccf237e298.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Dev/C++/vcpkg/downloads/mysql-mysql-server-mysql-8.0.34.tar.gz
-- Applying patch dependencies.patch
-- Applying patch install-exports.patch
-- Applying patch fix_dup_symbols.patch
-- Applying patch cross-build.patch
-- Using source at D:/Dev/C++/vcpkg/buildtrees/libmysql/src/sql-8.0.34-ccf237e298.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: D:/Dev/C++/vcpkg/buildtrees/libmysql/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\Dev\C++\vcpkg\buildtrees\libmysql\config-x64-windows-dbg-CMakeCache.txt.log
      D:\Dev\C++\vcpkg\buildtrees\libmysql\config-x64-windows-rel-CMakeCache.txt.log
      D:\Dev\C++\vcpkg\buildtrees\libmysql\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/libmysql/portfile.cmake:51 (vcpkg_cmake_configure)
============================================================================
The log files are attached:
[config-x64-windows-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/13278612/config-x64-windows-dbg-CMakeCache.txt.log)
[config-x64-windows-rel-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/13278615/config-x64-windows-rel-CMakeCache.txt.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13278618/config-x64-windows-out.log)
[config-x64-windows-err.log](https://github.com/microsoft/vcpkg/files/13278620/config-x64-windows-err.log)

Regards,
    Paul McGinnie

**Comment by zxffffffff** [comment]
on my mac m1, use static lib, maybe the same problem?

ld: Undefined symbols:
  _res_9_dn_expand, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)::$_0::operator()(__ns_rr const&) const in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
  _res_9_nclose, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
  _res_9_ninit, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
  _res_9_ns_initparse, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
  _res_9_ns_parserr, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
  _res_9_nsearch, referenced from:
      cdk::foundation::connection::detail::srv_list(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in libmysqlcppconn8-static.a[102](socket_detail.cc.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [sample-main/sample-main] Error 1

**Comment by dg0yt** [comment]
- The orignal issue was about the configuration step for cross build. This might be resolved since the 8.0.34 update (Nov 1).  @qlaiaqu mut check. Then this issue can be closed.
- The new Windows issue is caused by a working dir which triggers errors when it is used as a regular expression: `D:/Dev/C++/`. It is not a vcpkg bug, it is a problem for multiple ports, and mitigation is easy. @mcginnie must act.
- The new Mac issue is unclear. @zxffffffff You better report a new issue with all required information.
